src/Simplify.cpp

```diff
@@ -2040,6 +2071,26 @@ class Simplify : public IRMutator {
             // min(y, min(x, y)) -> min(x, y)
             expr = b;
         } else if (min_a &&
+                   broadcast_a_b &&
+                   broadcast_b ) {
+            // min(min(x, broadcast(y, n)), broadcast(z, n))) -> min(x, broadcast(min(y, z), n))
+            expr = mutate(Min::make(min_a->a, Broadcast::make(Min::make(broadcast_a_b->value, broadcast_b->value), broadcast_b->lanes)));
+        } else if (min_a &&
```

Comment:
I don't think you need these three cases if you add:

if (a.as<Broadcast>() && !b.as<Broadcast>()) {
  std::swap(a, b);
}

To the top of the visitor